### PR TITLE
Refactor End User Guide#Using developer workspaces + adapt to dashboard next

### PR DESCRIPTION
**Pull request by sleshchenko**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

It targets Che 7.26 (it's a first release where a new Dashboard is the the default)

It targets CRW 2.8 (2.7 still has old dashboard, the latest plan it pull a new Dashboard into CRW 2.8)

### Massive update for EuG
* cleaning and updating
* new content
* better navbar menu orientation
* brings additional sanity check to the whole EuG (attributes, headings, modularization, proper branding/cropped images/langage/style

### What does this PR do?
The main purpose of this PR is to adapt documentation to switch to Dashboard Next.
Apart from small self-descriptive fixes it has:
1. Refactoring the structure of `End User Guide#Using developer workspaces`, now it has:
- Configure a workspace using a devfile - describes the devfile structure ( the content is not reviewed but just moved )
- Creating a workspace from a remote devfile - describes factory URL
- Creating a workspace from code sample - describes how to create a workspace from Get Started (sample) + Create Custom Workspace (from template)
- Configuring a workspace - describes how to configure workspace, editor, plugins, projects...
- ... other existing pieces that I did not touch like Importing K8s application, Mounting a secret as a file...
See https://5fed87541c14ed21f992d502--eclipse-che-docs-pr.netlify.app/che-7/end-user-guide/workspaces-overview/
2. Rework Copy chectl login command. See https://5fed87541c14ed21f992d502--eclipse-che-docs-pr.netlify.app/che-7/end-user-guide/navigating-che-using-the-dashboard/#logging-in-to-eclipse-che-using-chectl_navigating-che-using-the-dashboard
3. Rework starting a workspace in debug mode (now it's verbose mode). See https://5fed87541c14ed21f992d502--eclipse-che-docs-pr.netlify.app/che-7/end-user-guide/starting-a-workspace-in-verbose-mode/

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18518

### Specify the version of the product this PR applies to.
7.26+

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc
Comment (sleshchenko) [inline modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc]: Probably here we should specify MacOS specific hotkey as well, like Command+Space?

Comment (benoitf) [reply modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc]: Should you use https://asciidoctor.org/docs/user-manual/#keyboard-shortcuts also ?

File: modules/end-user-guide/partials/proc_logging-in-to-che-server-using-cli.adoc
Comment (sleshchenko) [inline modules/end-user-guide/partials/proc_logging-in-to-che-server-using-cli.adoc]: For reviewers to easier identify if top-right corner is good enough description for it:

Screenshot_20201230_201402

File: antora-playbook.yml
Comment (sleshchenko) [inline antora-playbook.yml]: It does not seem to have the right value, more so it's not used anywhere. So, maybe it's just better to remove it.

Comment (MichalMaler) [reply antora-playbook.yml]: Yeah, I just double-checked it. No mention in the upstream anymore. Are you giving me a green to delete the `making-a-workspace-portable-using-a-devfile.adoc` file ? :)

Comment (sleshchenko) [reply antora-playbook.yml]: The property is removed in https://github.com/eclipse/che-docs/pull/1737/commits/70c03801c591bdc0d6361186bc0255bd33f22b15

Does `making-a-workspace-portable-using-a-devfile.ado` really exist?

**Comment by sleshchenko** [comment]
The script below helped me to detect that I lost:
- modules/end-user-guide/partials/assembly_creating-a-workspace-from-remote-devfile.adoc <- is already recovered in https://github.com/eclipse/che-docs/pull/1737/commits/52c8b55810c7e5cfa8fb57b4082e3e439cb52868
- proc_creating-a-workspace-using-chectl-and-a-local-devfile.adoc <- is recovered in https://github.com/eclipse/che-docs/pull/1737/commits/d2b8a158b50b7b199f287e852787994574dcba9e

In addition, it reports the following partials being not used:

```
!!! The following partials are not used in their modules.
!!! Remove them to fix the issue.

proc_installing-che-using-the-che-operator-in-openshift-4-web-console.adoc
assembly_using-a-visual-studio-code-extension-in-che.adoc
```

After I modified script just a bit, it's detected that the following page is not used as well: `xref:installing-che-locally.adoc`

<details>
<summary>detect_unused_partials.sh</summary>

```bash
#!/bin/bash

# Exit on any error
set -e

# This is a configuration parameters that can be che or crwctl
PROJECT_CONTEXT=${PROJECT_CONTEXT:-che}

SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
DOCS_PROJECT_PATH=$SCRIPT_DIR/..

# we need to move it to make find print relative paths
pushd "$DOCS_PROJECT_PATH/modules" > /dev/null
readarray -d '' modules < <(find . -mindepth 1 -maxdepth 1 -type d -print0)

unused_partials=""
for module in "${modules[@]}"
do
    pushd "$module" > /dev/null

    if [ ! -d "./partials" ]; then        
        # This module does not partials"
        popd > /dev/null
        continue
    fi

    readarray -d '' partials < <(find "partials" -name '*.adoc' -print0)
    for partial in "${partials[@]}"
    do
        #`../` instead of `partials/` is used in the documentation references
        partial=${partial#"partials/"}
        partial_with_che_context="${partial/che/{project-context\}}"
        partial_with_crw_context="${partial/${PROJECT_CONTEXT}/{project-context\}}"
        if ! grep -q -r "$partial" . && ! grep -q -r "$partial_with_che_context" . && ! grep -q -r "$partial_with_crw_context" . ; then
            unused_partials="$unused_partials\n$partial"
        fi
    done

    popd > /dev/null
done

popd > /dev/null

if [[ "$unused_partials" ]]; then
    echo "!!! The following partials are not used in their modules."
    echo "!!! Remove them to fix the issue."
    echo -e "${unused_partials}"
    exit 1
else
    echo "All partials have references in their the modules."
fi

```

</details>

P.S. I'm not pushing these scripts since it's needed to refactor `detect_unused_images.sh` to check unused all kind of files. Hope I will be able to contribute such piece later.

File: antora-playbook.yml
Comment (themr0c) [reply antora-playbook.yml]: It's probably an old name. We have quite some leftovers like this. Any help to detect and fix them is very appreciated!

**Comment by themr0c** [comment]
@sleshchenko Does the script take into account the includes from files in all relevant directories ? (pages, partials, examples)

**Comment by sleshchenko** [comment]
> @sleshchenko Does the script take into account the includes from files in all relevant directories ? (pages, partials, examples)

@themr0c yes, it look everywhere in the current module folder

> grep -q -r "$partial" .

I also re-checked reported items manually with full-text-search and it seems to be not used anywhere.

File: antora-playbook.yml
Comment (sleshchenko) [reply antora-playbook.yml]: The following script:

<details>
<summary>detect_unused_properties</summary>

```bash
#!/bin/bash

# Exit on any error
set -e

# This is a configuration parameters that can be che or crwctl
PROJECT_CONTEXT=${PROJECT_CONTEXT:-che}

SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
DOCS_PROJECT_PATH=$SCRIPT_DIR/..

unused_pages=""

    readarray -d '' pages < <(cat antora-playbook.yml | yq -j '.asciidoc.attributes | keys[] | ( . + "\u0000")')

    for page in "${pages[@]}"
    do
        #`../` instead of `pages/` is used in the documentation references
        # page=${page//pages\//xref:}
        echo "Search for $page in $(pwd)"
        page="{$page}"
        if ! grep -q -r "$page" . ; then
            unused_pages="$unused_pages\n$page"
        fi
    done

if [[ "$unused_pages" ]]; then
    echo "!!! The following pages are not used in their modules."
    echo "!!! Remove them to fix the issue."
    echo -e "${unused_pages}"
    exit 1
else
    echo "All pages have references in their the modules."
fi

```

</details>

detected the following unused properties:

```
{experimental}
{favicon}
{icons}
{link-advanced-configuration-options-for-the-che-server}
{link-building-a-custom-plug-in-registry}
{link-viewing-the-state-of-the-cluster-deployment-using-openshift-4-cli-tools}
{ocp4-ver}
{prod-prev-ver-major}
{prod-url-secure}
{prod-ver-major}
```

When I will have time - I'll refactor detect_unused_images to take into account properties as well.

**Comment by MichalMaler** [comment]
@rkratky 
Guys, we would like to refactor this section to have the content in better shape, because if we mix creating from remote devfile (which has a lot of pieces) + others, it will be hard to navigate.

From Fabrice:
We can have 4 nested levels starting with CRW 2.6. But I would say that's not very desirable, too much nesting is probably a good indicator we need to restructure in a simpler way
  
"Creating a workspace" becomes probably an empty container for level 4 items, so a flatter hierarchy stopping at level 3 could be more useful, like: # Creating a workspace using a sample

1. Creating a workspace using a remote devfile
2. Creating a workspace using a local devfile
3. Configuring an existing workspace
 
(NB: ordering 1. sample, 2. remote devfile, 3. local devfile, because that's how the learning path looks like: you use a sample to discover the product, an existing devfile for daily work, a local devfile occasionally when you are more advanced and want to modify a devfile.)

**Review by MichalMaler — APPROVED** [review]
Approving this stage. Merge this PR with the 7.25 release.

**Comment by MichalMaler** [comment]
@themr0c Fabrice, please, can I ask you kindly for an additional sanity check?

File: modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc
Comment (sleshchenko) [inline modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc:19]: The content above should be updated accordingly 
```
Every stack includes a sample codebase, which is defined by the devfile of the stack. This section explains how to create a workspace from this code sample in a sequence of three procedures.
. Creating a workspace from the user dashboard:
.. Using the xref:creating-a-workspace-from-get-started-view-of-user-dashboard_{context}[Get Started view].
.. Using the xref:creating-a-workspace-from-custom-workspace-view-of-user-dashboard_{context}[Custom Workspace view].
. xref:changing-the-configuration-of-an-existing-workspace_{context}[Changing the configuration of the workspace] to add code sample.
. xref:running-an-existing-workspace-from-the-user-dashboard_{context}[Running an existing workspace from the user dashboard].
```

**Review by sleshchenko — COMMENTED** [review]
I've reviewed `Creating a workspace from a remote devfile using the dashboard` and `Creating a workspace from code sample` articles. Will continue tomorrow

File: modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc
Comment (sleshchenko) [reply modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc:19]: Is fixed in https://github.com/eclipse/che-docs/pull/1737/commits/a05793bc6cea3445add584ce791ca2bb641fc802
@MichalMaler please verify

**Review by sleshchenko — COMMENTED** [review]
I'm done my review for `Using developer workspace`, added some fixes, improvements which probably needs to be reviewed )

File: modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc
Comment (MichalMaler) [reply modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc:19]: Done

**Comment by MichalMaler** [comment]
Some little typos + Upstream master rebase

**Review by MichalMaler — APPROVED** [review]

**Comment by themr0c** [comment]
This branch need a rebase.

**Comment by sleshchenko** [comment]
Rebased against master

**Comment by MichalMaler** [comment]
Michaylo already did 7.25 snapshots. So this PR is ready to be merged as one of the first PRs for Che 7.26.
@themr0c @rkratky @boczkowska Can I ask you for additional review, since this PR is kinda big?

**Comment by themr0c** [comment]
`master` is our publication branch. It is intended for the *released* version of Che.

**Comment by themr0c** [comment]
@sleshchenko what is the target for this PR now? I know it has moved a lot, so checking another time: 7.25 (dashboard agnostic) or 7.26 (only valid for dashboard next)?

**Comment by sleshchenko** [comment]
@themr0c the target for this PR is 7.26, where a new dashboard is available as the only option https://github.com/eclipse/che/pull/18549

**Review by benoitf — APPROVED** [review]

**Comment by themr0c** [comment]
https://github.com/eclipse/che/releases/tag/7.26.0 being out, it's time to have our second TW read.

File: modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc
Comment (yhontyk) [inline modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc]:
```suggestion
* Select your workspace by clicking its name. This will link you to the workspace's *Overview* tab.
```

File: modules/end-user-guide/partials/assembly_navigating-che-using-the-dashboard.adoc
Comment (yhontyk) [inline modules/end-user-guide/partials/assembly_navigating-che-using-the-dashboard.adoc]:
```suggestion
The *Dashboard* is accessible on your cluster from a URL such as `pass:c,a,q[{prod-url}/dashboard]`.
```

**Review by yhontyk — APPROVED** [review]
The build looks good to me, there are many vale errors that are out of the scope of this PR, I'll work on them separately. 
Approving.

@themr0c linkchecker fails on to URLs which seem like false positives to me, shall we add them to exceptions?

**Review by MichalMaler — APPROVED** [review]

**Comment by MichalMaler** [comment]
Will fix the VALE errors now.

**Comment by MichalMaler** [comment]
The VALE check needs to be disabled for this PR. The more I was trying to fix everything bad in EUG, the more I am touching files that are really nasty or a part of the Extensions and Contribution guide. I fixed 84 VALE errors + some additional ones, which I found and fixed using the search function. This PR does not spoil anything, just pointing to the sections that are not VALE friendly.
All originally affected docs were tuned and fixed.

**Comment by yhontyk** [comment]
> The VALE check needs to be disabled for this PR. The more I was trying to fix everything bad in EUG, the more I am touching files that are really nasty or a part of the Extensions and Contribution guide. I fixed 84 VALE errors + some additional ones, which I found and fixed using the search function. This PR does not spoil anything, just pointing to the sections that are not VALE friendly.
> All originally affected docs were tuned and fixed.

The vale checks are not blocking merging as of now so feel free to merge this PR if you find it ready to be merged. Please do file those vale errors in some issue so we can address them. =\

**Comment by MichalMaler** [comment]
https://github.com/sleshchenko/che-docs/pull/2 is another addition to this PR, but its content suits the 2.8 CRW release.

File: README.md
Comment (themr0c) [inline README.md:2]: Is the # a new way to load a factory?

Comment (benoitf) [reply README.md:2]: yes https://github.com/eclipse/che/issues/18165

Comment (sleshchenko) [reply README.md:2]: but note that it's not supported in CRW yet

**Comment by themr0c** [comment]
We miss redirections for previous URLs:

* http://0.0.0.0:4000/che-7/end-user-guide/making-a-workspace-portable-using-a-devfile/
* http://0.0.0.0:4000/che-7/end-user-guide/creating-and-configuring-a-new-workspace/
* http://0.0.0.0:4000/che-7/end-user-guide/creating-a-workspace-by-importing-source-code-of-a-project/
* http://0.0.0.0:4000/che-7/end-user-guide/starting-a-che-workspace-in-debug-mode/
* http://0.0.0.0:4000/che-7/end-user-guide/restarting-a-che-workspace-in-debug-mode-after-start-failure/

**Comment by themr0c** [comment]
Adding redirections for old URLs: https://github.com/sleshchenko/che-docs/pull/3

File: README.md
Comment (themr0c) [reply README.md:2]: Yes, we will make sure the content of this PR doesn't go into the 7.26.x branch that will be used for CRW 2.7 docs.

**Pull request merged**
